src/ui/widgets/tui_dirlist_detailed.rs

```diff
@@ -80,113 +80,197 @@ fn print_entry(
     (x, y): (u16, u16),
     drawing_width: usize,
 ) {
-    let name = entry.label();
-    let name_width = name.width();
-
-    match entry.metadata.file_type() {
-        FileType::Directory => {
-            buf.set_stringn(x, y, name, drawing_width, style);
-            if name_width > drawing_width {
-                buf.set_string(x + drawing_width as u16 - 1, y, ELLIPSIS, style);
-            }
-        }
-        FileType::Symlink(_) => {
-            if drawing_width < SYMLINK_WIDTH {
-                return;
-            }
-            let file_drawing_width = drawing_width - SYMLINK_WIDTH;
-
-            print_file_name(buf, (x, y), name, style, file_drawing_width);
+    let size_string = match entry.metadata.file_type() {
+        FileType::Directory => String::from(""),
+        FileType::File => format::file_size_to_string(entry.metadata.len()),
+    };
+    let symlink_string = match entry.metadata.link_type() {
```

Comment:
This part probably doesn't need to be a String, and can just be a static str, unless you're planning to do more with it later.
```rust
    let symlink_string = match entry.metadata.link_type() {
        LinkType::Normal => "",
        LinkType::Symlink(_) => "-> ",
    };
```